pkg/engine/lifecycle_test.go

```diff
@@ -1214,7 +1234,7 @@ func TestRefreshInitFailure(t *testing.T) {
 	// Refresh DOES fail, causing the new initialization error to appear.
 	//
 	refreshShouldFail = true
-	p.Steps = []TestStep{{Op: Refresh, ExpectFailure: true}}
+	p.Steps = []TestStep{{Op: Refresh, SkipPreview: true, ExpectFailure: true}}
```

Comment:
This test previously worked because it tested that the preview mutated the in-memory snapshot. While this is technically correct, a more correct test would be to test that an update correctly persists the init failure to the journal.

Reply:
Can you elaborate? The snapshot we verify below is created based on the journal and the refreshed base snapshot.

Reply:
Previous to this commit, we were using the same base snapshot for both the preview and the update. Since the refresh mutates the base snapshot in-place, the base snapshot that comes out of the preview having been mutated.

There are two cases where this causes trouble for lifecycle tests:

1. Any resource that is replaced in a preview is marked pending delete in the base snapshot. If a preview and update share the same base snapshot, an update will observe pending deletions in the base snapshot because the preview has mutated them.
2. Any resource that is refreshed in a preview mutates the snapshot so that it represents the refreshed state.

Both of these are trouble. This is why I am explicitly cloning the snapshot in `(*TestPlan).Run`. For this particular case, the Journal (when Snapped) applies no snapshot mutations on top of a base snapshot that has been mutated by refresh. Therefore, the test passes - the returned snapshot here has exactly the set of changes that we want out of this test. However, the test was only doing a preview, because the preview immediately failed and an update never ran.